Dart: generate omitted documentation of properties

pwrobeldev opened the pull request:
When users want to document a property
in LIME files, they can use structured
comments, which consist of the general
information about property and annotations
related to getters and setters.

In the case of Dart generator, the output
contained only comments related to getters
and setters. The general information about
property was not stored in the output files.

This change adjusts 'DartRedirectProperty'
mustache template to output also the general
information about property if present.

The behaviour in the case of Dart generator
has been aligned with C++ and Java generators.
The general information about property is added
to comments of getter and setter.

kcherneha commented:
Across almost all files I see redundant new lines, could you please remove them?
As well as {{escape ""}} - not sure if needed, 

```

  /// Property comment
  /// Description of: String prop.
  --- here is empty line - I don't think we need it?
  /// Getter comment
  @OnPropertyInClass
```

pwrobeldev commented:
> Across almost all files I see redundant new lines, could you please remove them? As well as {{escape ""}} - not sure if needed,
> 
> ```
> 
>   /// Property comment
>   /// Description of: String prop.
>   --- here is empty line - I don't think we need it
>   /// Getter comment
>   @OnPropertyInClass
> ```

Fixed.

My initial idea was to separate the field description from the comment that describes getter. Because after the proposed change the general part of property comment is a part of getter description in Dart. However, if this is the requirement, then I am okay with it.

In C++/Java both getter and setter have the part related to the general comment about property. If needed, this can be also added to setter in Dart -- the change will be 1 line. 

Because right now the behavior is as follows:
- getter in Dart contains `@get` comment as well as the general comment
- setter in Dart contains only `@set` comment

pwrobeldev commented:
Also, regarding the empty lines -- Smoke tests do not require updating files when only whitespaces change. Therefore, my change accumulated also changes in whitespaces that were done long time ago. 

Please check: `private fun ignoreUnimportantDifferences(text: String)` function that is present in SmokeTest.kt.

kcherneha commented:
> > Across almost all files I see redundant new lines, could you please remove them? As well as {{escape ""}} - not sure if needed,
> > ```
> > 
> >   /// Property comment
> >   /// Description of: String prop.
> >   --- here is empty line - I don't think we need it
> >   /// Getter comment
> >   @OnPropertyInClass
> > ```
> 
> Fixed.
> 
> My initial idea was to separate the field description from the comment that describes getter. Because after the proposed change the general part of property comment is a part of getter description in Dart. However, if this is the requirement, then I am okay with it.
> 
> In C++/Java both getter and setter have the part related to the general comment about property. If needed, this can be also added to setter in Dart -- the change will be 1 line.
> 
> Because right now the behavior is as follows:
> 
> * getter in Dart contains `@get` comment as well as the general comment
> * setter in Dart contains only `@set` comment

Ideally, Swift, Java and Dart (from comments perspective) should be aligned as much as possible. If it doesn't bring redundancy and good for readability - you can add it as well.

Hsilgos commented on gluecodium/src/test/resources/smoke/attributes/output/dart/lib/src/smoke/attributes_with_comments.dart:
Is here any changes?

pwrobeldev replied:
Only line 21: `/// Property comment`

Hsilgos commented:
> Across almost all files I see redundant new lines, could you please remove them? As well as {{escape ""}} - not sure if needed,
> 
> ```
> 
>   /// Property comment
>   /// Description of: String prop.
>   --- here is empty line - I don't think we need it
>   /// Getter comment
>   @OnPropertyInClass
> ```

Empty lines were introduced not by this change. Some empty lines should be removed with https://github.com/heremaps/gluecodium/pull/1553 but generally speaking either we have empty lines or more clean mustache files.
